pkg/common/telemetry/call.go

```diff
@@ -0,0 +1,38 @@
+package telemetry
+
+import "time"
+
+type CallCounter struct {
+	metrics Metrics
+	err     *error
+	key     []string
+	labels  []Label
+	start   time.Time
+}
+
+func StartCall(metrics Metrics, err *error, key string, keyn ...string) *CallCounter {
+	return &CallCounter{
+		metrics: metrics,
+		err:     err,
```

Comment:
Is it safe to give this error pointer directly to the telemetry package? You would have to be sure not to update error after having called Done()...?

Perhaps there are some other options for how to handle an error condition, considering my comment above about error counter emission?